client/app/account/billing/main/history/downloadBills/billing-main-history-downloadBills.controller.js

```diff
@@ -0,0 +1,108 @@
+angular.module('Billing').controller('BillingMainHistoryDonwloadBillsCtrl', class BillingMainHistoryDonwloadBillsCtrl {
+  constructor($q, $translate, $uibModalInstance, Alerter, OvhApiMe) {
+    this.$q = $q;
+    this.$translate = $translate;
+    this.$uibModalInstance = $uibModalInstance;
+    this.Alerter = Alerter;
+    this.OvhApiMe = OvhApiMe;
+  }
+
+  $onInit() {
+    this.archiveName = `${this.$translate.instant('billing_main_bills_tab')}_${moment().format('YYYY_MM_DD_hh_mm_ss')}`;
+    this.billsCount = null;
+    this.apiV7Request = this.OvhApiMe.Bill().v7().query().sort('date', 'DESC');
+  }
+
+  hasError() {
+    return !!this.error || !!this.apiError;
+  }
+
+  isDateFromBeforeDateTo() {
+    this.error = null;
+    if (!this.searchTo) {
+      return true;
+    }
+
+    const searchFrom = moment(this.searchFrom, 'YYYY-MM-DD');
+    const searchTo = moment(this.searchTo, 'YYYY-MM-DD');
+
+    if (moment(searchFrom).isAfter(searchTo)
+      || moment(searchFrom).isSame(searchTo)) {
+      this.error = this.$translate.instant('billing_main_bill_download_date_from_after_date_to');
+      return false;
+    }
+
+    return true;
+  }
+
+  isDateRangeValid() {
+    if (!this.searchTo) {
+      return true;
+    }
+
+    const searchFrom = moment(this.searchFrom, 'YYYY-MM-DD');
+    const searchTo = moment(this.searchTo, 'YYYY-MM-DD');
+
+    if (moment(searchTo).diff(searchFrom, 'month') >= 1) {
+      this.error = this.$translate.instant('billing_main_bill_download_date_range_invalid');
+      return false;
+    }
+
+    return true;
+  }
+
+  getBills() {
+    this.downloading = true;
+    this.apiError = null;
+
+    if (!this.searchTo || moment(this.searchTo, 'YYYY-MM-DD').isSame(moment(this.searchFrom, 'YYYY-MM-DD'))) {
+      this.apiV7Request = this.apiV7Request
+        .addFilter('date', 'ge', moment(this.searchFrom, 'YYYY-MM-DD').format('YYYY-MM-DD'))
```

Comment:
There is two formatting, one for the moment js deprecated message, one for the api route param requirements